### PR TITLE
Add unit tests for io.restassured.http.Header

### PR DESCRIPTION
**Pull request by petersear**
I've written some tests for the methods in `io.restassured.http.Header` with the help of [Diffblue Cover](https://www.diffblue.com/opensource).

Hopefully, these tests will help detect any regressions caused by future code changes. If you would find it useful to have additional tests written for this repository, I would be more than happy to look at other classes in a subsequent PR.

### CODE REVIEW

**Review by freddytuxworth — APPROVED** [review]

**Review by paulkennethkent — APPROVED** [review]

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    assertThat(header3.hasSameNameAs(header)).isFalse();
+  }
+
+  @Test
+  public void testEquals() {
+    final Header o = new Header("foo", "bar");
+
+    final Header header = o;
+    assertThat(header.equals(o)).isTrue();
```
Comment (Braavos6) [inline rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: for the test `if (this == o) return true;` I would do `assertThat(o.equals(o)).isTrue();`
Rather than creating a copy of the object it would be better to use the same object to specific that they are equal. As this would include the reference in memory not just the values of the object

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    assertThat(header2.equals(o)).isTrue();
+
+    final Header header3 = new Header("", "");
+    assertThat(header3.equals(o)).isFalse();
+
+    final Header header4 = new Header("foo", "");
+    assertThat(header4.equals(o)).isFalse();
+
+    final Header header5 = new Header("", "bar");
+    assertThat(header5.equals(o)).isFalse();
```
Comment (Braavos6) [inline rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: Also you could have passed a `null` value for the test `if (o == null || getClass() != o.getClass()) return false;`... for the branch coverage ;)

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    assertThat(header2.hasSameNameAs(header)).isTrue();
+
+    final Header header3 = new Header("bar", "baz");
+
+    assertThat(header3.hasSameNameAs(header)).isFalse();
+  }
+
+  @Test
+  public void testEquals() {
+    final Header o = new Header("foo", "bar");
```
Comment (Braavos6) [inline rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: If the object `o` is the constant wouldn't it be better to call `.equals` on this object? and the parameter to be the one that changes?

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java (hunk 1 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: But `header` is just a reference to `o`, not a separate object and does not allocate more memory.
I somehow feel the intention is clearer if creating a new reference and giving the appearance of comparing two items.

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java (hunk 3 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: Although it works the other way around this seems to make more sense, thanks.
Done.

File: rest-assured/src/test/java/io/restassured/http/HeaderTest.java (hunk 2 above)
Comment (petersear) [reply rest-assured/src/test/java/io/restassured/http/HeaderTest.java]: I've added two new tests to exercise the `o == null` and `getClass() != o.getClass()` conditions.

**Review by Braavos6 — APPROVED** [review]
